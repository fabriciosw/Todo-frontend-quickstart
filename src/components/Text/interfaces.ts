import React from 'react';

export interface IText {
  children: React.ReactNode;
  as:
    | 'p'
    | 'small'
    | 'b'
    | 'i'
    | 'em'
    | 'mark'
    | 'del'
    | 'ins'
    | 'sub'
    | 'sup'
    | 'span'
    | 'strong'
    | 'h1'
    | 'h2'
    | 'h3'
    | 'h4'
    | 'h5'
    | 'h6';
  size?: string;
  weight?: 400 | 500 | 700;
  color?: string;
  className?: string;
}
