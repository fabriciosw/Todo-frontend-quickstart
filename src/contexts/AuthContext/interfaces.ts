import React from 'react';
import firebase from '../../config/firebase';

export interface IAuthContext {
  currentUser: firebase.User | null;
  signup: (displayName: string, email: string, password: string) => Promise<firebase.auth.UserCredential>;
  login: (email: string, password: string) => Promise<firebase.auth.UserCredential>;
  logout: () => Promise<void>;
  resetPassword: (email: string) => Promise<void>;
  updateEmail: (email: string) => Promise<void>;
  updatePassword: (email: string) => Promise<void>;
}

export interface IAuthProvider {
  children: React.ReactElement;
}
