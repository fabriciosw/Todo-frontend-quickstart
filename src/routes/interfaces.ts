import React from 'react';

export interface IRoute {
  path: string;
  component: React.FunctionComponent;
  public?: boolean;
  title: string;
}
