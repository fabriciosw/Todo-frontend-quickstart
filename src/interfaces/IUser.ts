export interface IUser {
  createdAt: Date;
  first_name: string;
  last_name: string;
  email: string;
  id: string;
}
