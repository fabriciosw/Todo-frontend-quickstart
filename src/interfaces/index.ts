export * from './IData';
export * from './IUser';
