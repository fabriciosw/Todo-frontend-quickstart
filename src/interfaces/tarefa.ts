export interface ITarefa {
  id: number;
  title: string;
  description: string;
  complete: boolean;
}
