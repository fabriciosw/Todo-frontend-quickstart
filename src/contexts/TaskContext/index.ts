import { createContext, useContext } from 'react';
import { ITarefa } from '../../interfaces/tarefa';

type Props = {
  tarefas: ITarefa[];
  setTarefas: (x: ITarefa[]) => void;
};

export const Tarefa = createContext<Props>({} as Props);

export const Tarefas = (): Props => useContext(Tarefa);
