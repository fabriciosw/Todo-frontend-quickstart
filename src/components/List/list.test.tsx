import React from 'react';
import { render } from '@testing-library/react';
import Lista from './index';
import { Tarefa } from '../../contexts/TaskContext/index';
import { ITarefa } from '../../interfaces/tarefa';

it('must match snapshot', () => {
  const tarefas: ITarefa[] = [
    {
      id: 1,
      title: 'first task',
      description: 'big desc',
      complete: false,
    },
    {
      id: 2,
      title: 'second task',
      description: 'small =D',
      complete: true,
    },
  ];
  const setTarefas = jest.fn();

  const { container } = render(
    <Tarefa.Provider value={{ tarefas, setTarefas }}>
      <Lista taskStatus={undefined} />
    </Tarefa.Provider>
  );

  expect(container).toMatchSnapshot();
});
