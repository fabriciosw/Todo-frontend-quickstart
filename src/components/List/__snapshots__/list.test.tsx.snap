// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must match snapshot 1`] = `
<div>
  <aside>
    <div
      class="tarefa"
    >
      <div
        class="dados"
      >
        <p>
          first task
        </p>
        <span>
          big desc
        </span>
      </div>
      <div
        class="opcoes"
      >
        <img
          alt="terminar"
          src="finish.jpg"
        />
        <img
          alt="deletar"
          src="Trash.png"
        />
      </div>
    </div>
    <div
      class="tarefa finalizado"
    >
      <div
        class="dados"
      >
        <p>
          second task
        </p>
        <span>
          small =D
        </span>
      </div>
      <div
        class="opcoes"
      >
        <img
          alt="deletar"
          src="Trash.png"
        />
      </div>
    </div>
  </aside>
</div>
`;
